android/sdk/src/main/cpp/app/organicmaps/sdk/bookmarks/data/PredefinedColors.cpp

```diff
@@ -0,0 +1,40 @@
+#include "PredefinedColors.hpp"
+
+#include "kml/types.hpp"
+
+#include <array>
+
+extern "C"
+{
+JNIEXPORT jintArray JNICALL
+Java_app_organicmaps_sdk_bookmarks_data_PredefinedColors_nativeGetPredefinedColors(JNIEnv * env, jclass)
+{
+  std::array<jint, static_cast<size_t>(kml::PredefinedColor::Count)> colors{};
+  for (size_t i = 0; i < static_cast<size_t>(kml::PredefinedColor::Count); ++i)
+    colors[i] = static_cast<jint>(kml::ColorFromPredefinedColor(static_cast<kml::PredefinedColor>(i)).GetARGB());
+
+  jintArray jColors = env->NewIntArray(colors.size());
+  env->SetIntArrayRegion(jColors, 0, static_cast<jsize>(colors.size()), colors.data());
+  return jColors;
+}
+}
+
+namespace
+{
+static JNINativeMethod methods[] = {
+    {"nativeGetPredefinedColors", "()[I",
+     reinterpret_cast<void *>(&Java_app_organicmaps_sdk_bookmarks_data_PredefinedColors_nativeGetPredefinedColors)},
+};
+}
+
+namespace predefined_colors
+{
+jint registerNativeMethods(JNIEnv * env)
+{
+  jclass const clazz = env->FindClass("app/organicmaps/sdk/bookmarks/data/PredefinedColors");
+  if (clazz == nullptr)
+    return JNI_ERR;
+
+  return env->RegisterNatives(clazz, methods, sizeof(methods) / sizeof(methods[0]));
+}
```

Comment:
What are the pros and cons of introducing a different approach to using/calling JNI functions compared to the one already used in master?

Reply:
```
The built-in dynamic JNI linking is working only in Android 12+, the explicit registration with JNI RegisterNatives is strictly required for running on Android versions 8-11. The annotation is ignored on Android 7-.
```
https://developer.android.com/reference/dalvik/annotation/optimization/FastNative

Reply:
It is required for Android11- for `@FastNative` call
It could also be used to provide different name for a native function, but I left the same name
On Android11- this function will be mapped by RegisterNatives call, on Android12+, using the standard jni runtime magic